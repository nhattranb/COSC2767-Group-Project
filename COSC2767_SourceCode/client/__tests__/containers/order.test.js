/*
 *   RMIT University Vietnam
 *   Course: COSC2767 Systems Deployment and Operations
 *   Semester: 2024C
 *   Assessment: Assignment 2
 *   Author: Le Dinh Ngoc Quynh
 *   ID: s3791159
 *   Created  date: 25 Dec 2024
 *   Last modified: 25 Dec 2024
 */

import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Order from '../../app/containers/Order';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Order component', () => {
  let store;

  const initialState = {
    order: {
      orders: [],
      isLoading: false,
      currentOrder: null
    },
    authentication: {
      authenticated: true
    }
  };

  beforeEach(() => {
    store = mockStore(initialState);
    store.dispatch = jest.fn();
  });

  it('renders correctly', () => {
    const stateWithUser = {
      ...initialState,
      account: {
        user: {}
      }
    };
    store = mockStore(stateWithUser);

    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Order />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders loading state correctly', () => {
    const loadingState = {
      ...initialState,
      order: {
        ...initialState.order,
        isLoading: true
      }
    };
    store = mockStore(loadingState);

    const loadingStateWithUser = {
      ...loadingState,
      account: {
        user: {}
      }
    };
    store = mockStore(loadingStateWithUser);

    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Order />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with orders correctly', () => {
    const ordersState = {
      ...initialState,
      order: {
        ...initialState.order,
        orders: [
          {
            _id: '1',
            total: 100,
            created: new Date(),
            products: []
          }
        ]
      }
    };
    store = mockStore(ordersState);

    const ordersStateWithUser = {
      ...ordersState,
      account: {
        user: {}
      }
    };
    store = mockStore(ordersStateWithUser);
    
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Order />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
}); 