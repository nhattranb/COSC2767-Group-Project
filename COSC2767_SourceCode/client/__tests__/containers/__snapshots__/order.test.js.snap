// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="product-dashboard"
  >
    <div
      class="page-404"
    >
      The page you are looking for was not found.
    </div>
  </div>
</DocumentFragment>
`;

exports[`Order component renders loading state correctly 1`] = `
<DocumentFragment>
  <div
    class="product-dashboard"
  >
    <div
      class="page-404"
    >
      The page you are looking for was not found.
    </div>
  </div>
</DocumentFragment>
`;

exports[`Order component renders with orders correctly 1`] = `
<DocumentFragment>
  <div
    class="product-dashboard"
  >
    <div
      class="page-404"
    >
      The page you are looking for was not found.
    </div>
  </div>
</DocumentFragment>
`;
